code/parse/sexp.cpp

```diff
@@ -20448,6 +20448,7 @@ void sexp_debug(int node)
 	node = CDR(node); 
 	Assertion (node >= 0, "No message defined in debug SEXP"); 
 	id = CTEXT(node);
+	strcpy_s(temp_buf, id);
```

Comment:
I suppose.  I've modified the sexp function to be a lot more like `sexp_set_death_message`, which it is conceptually close to.